lib/eco/treemanager.py

```diff
@@ -628,7 +629,14 @@ def jump_to_error(self, parser):
 
     def key_shift_ctrl_z(self):
         self.log_input("key_shift_ctrl_z")
-        if self.get_max_version() > self.version:
+        try:
+            i = self.undo_snapshots.index(self.version)
+            if i == len(self.undo_snapshots) - 1:
+                return
+            undo_amount = self.undo_snapshots[i+1] - self.undo_snapshots[i]
+        except ValueError:
```

Comment:
Not massively huge. As big as how many undo operations you have in your document. The longer you tube, the bigger. You are more likely to run out of memory than making running over `undo_snapshots` slow. I just think it's bad programming to iterate over a list twice when it can be done in one.